Block/Analytics.php

```diff
@@ -19,22 +19,24 @@ class Analytics extends Template
     public $helper;
 
     /**
-     * @param Context                            $context
-     * @param \Metrilo\Analytics\Helper\Data     $helper
-     * @param \Metrilo\Analytics\Model\Analytics $dataModel
-     * @param \Magento\Customer\Model\Session    $session
-     * @param array                              $data
+     * @param Context                                           $context
+     * @param \Magento\Framework\App\Action\Context             $actionContext
+     * @param \Metrilo\Analytics\Helper\Data                    $helper
+     * @param \Metrilo\Analytics\Model\Analytics                $dataModel
+     * @param \Metrilo\Analytics\Helper\Events\ProductViewEvent $productViewEvent,
+     * @param array                                             $data
      */
     public function __construct(
         Context $context,
-        \Metrilo\Analytics\Helper\Data $helper,
-        \Metrilo\Analytics\Model\Analytics $dataModel,
-        \Magento\Customer\Model\Session $session,
+        \Magento\Framework\App\Action\Context             $actionContext,
+        \Metrilo\Analytics\Helper\Data                    $helper,
+        \Metrilo\Analytics\Helper\Events\ProductViewEvent $productViewEvent,
```

Comment:
Do we really want this to be a helper? We're going to have to add ALL types of events here then. Maybe a model is better? What do you think?